src/gtfs/read.rs

```diff
@@ -957,6 +941,42 @@ pub fn read_routes<P: AsRef<path::Path>>(path: P, collections: &mut Collections)
     Ok(())
 }
 
+pub fn get_validity_period(
+    calendars: &CollectionWithId<objects::Calendar>,
+) -> Option<objects::ValidityPeriod> {
+    let dates = calendars.values().fold(BTreeSet::new(), |acc, c| {
+        acc.union(&c.dates).cloned().collect()
+    });
+
+    if dates.is_empty() {
+        return None;
+    }
+
+    Some(objects::ValidityPeriod {
+        start_date: *dates.iter().next().unwrap(),
+        end_date: *dates.iter().next_back().unwrap(),
```

Comment:
It will fail with only one day.

Reply:
no it won't. I added a test for this case

Reply:
ha, my bad, was thinking it was the same iterator, but there is one for each.

Thus, what's wrong with front() and back()?

Reply:
OK, my bad.